### ISSUE TITLE
Autofire doesn't work with scopes

### ISSUE THREAD

**Issue by Henriquekill9576-taken** [tgstation/TerraGov-Marine-Corps#3292]
## Testmerges:

Round ID: 1779
BYOND Version: 512.1488
Server revision compiled on: 
Master commit: 7b60722339cd03a814401bf2f051beadb6ec2f43
The following pull requests are currently test merged:
#3283: 'New spacemandmm release' by spookydonut at commit 18cf9ff355
#3284: 'Minor APC cleanup' by Rohesie at commit 53d2fff134
#3287: 'unfuck shutter element' by spookydonut at commit 5bf010099c
#3280: 'Reverts IS to old storage' by Kuro-ai at commit ca9302db5a
#3275: 'Marine Snow Slowdown reduced from +75% move speed penalty to +50% move speed penalty' by BurgerLUA at commit 44cb26c22e
#3251: 'Reworks the Canterbury Cream Egg Shuttle' by BurgerLUA at commit e8cf1b17e9
#3248: 'HUD fix' by Rohesie at commit f60c1696b0
#3272: 'remove br memes' by Kuro-ai at commit d643d21bc7
#3273: 'Buffs machete,nerfs weed nodes' by Kuro-ai at commit 49fd6a8e6b
#3282: 'Respawn Game tab fix' by Rohesie at commit 2fdb1492e4
#3281: 'Riot Armor Nerf' by Isy232 at commit 605c6f613a
#3274: 'T64 DMR' by David-Stormwell-TGCM at commit 9b480e0d0c

Local commit: d5718775b2058d7f1926ac92fef1b56471f52f5a
Server tools version: /tg/station 13 Server v3.2.6.2

## Reproduction:

1. Get a gun with autofire and put a scope in it
2. Try to use the autofire while using the scope

**Comment by Hughgent** [comment]
Current Master.

Turns out you CAN use autofire with a scope, but only if you click within your regular 7 tile view range.

Something to do with 'CLICKCATCHER_PLANE` @Rohesie

**Comment by Rohesie** [comment]
```
/datum/component/automatic_fire/proc/process_shot()
switch(get_dist(shooter, target))
	if(8 to INFINITY) //Can technically only go as far as 127 right now.
		stop_autofiring() //Elvis has left the building.
```
It wasn't designed with scopes in mind. That said, it can be changed to match the user client's `view` value, but it should also take into account tile offsets and changed client eyes. Not trivial, but doable.

**Comment by davidchan13** [comment]
Can confirm still autofire does not work with scope unless you click on a tile you could see before scoping.

**Comment by psykzz** [comment]
This should've been fixed with #4006 let me know if it comes up again
